### PR TITLE
709 jacoco path prefix

### PR DESCRIPTION
**Pull request by koschke**
We can now import JaCoCo coverage data if the source code of a Java project is distributed in multiple directories (e.g., `src/main/java`, `src-gen/main/java`, or  `src/test/java`). We are using only the main type of a file for the source-code range index.

This fixes issue #709.

### CODE REVIEW

**Review by koschke — COMMENTED** [review]
Looks good to me. Can be reviewed.

File: Assets/SEE/Game/City/EdgeMeshScheduler.cs [hunk 1]
```diff
@@ -187,7 +186,8 @@ public void OnNext(ChangeEvent value)
             if (value is EdgeEvent { Change: ChangeType.Addition } edgeEvent)
             {
                 // If this is an added edge, we are going to need to turn it into a mesh.
-                LoadingSpinner.Show(LoadingText);
+                // FIXME: Disabled temporarily. Do we want to activate it again?
```
[the remark this answers is not in the document]
Comment (koschke) [reply Assets/SEE/Game/City/EdgeMeshScheduler.cs]: Fixed.

File: Assets/SEE/DataModel/DG/IO/JaCoCoImporter.cs [hunk 2]
```diff
@@ -322,34 +346,169 @@ private static string XMLSourcePosition(string filepath, XmlReader xmlReader)
...
+            if (simpleName == typeAccordingToFilename)
             {
-                return sourceFilename;
+                // It is a main type.
+                return outerMostType;
             }
             else
             {
-                return qualifiedClassName.Remove(lastSeparatorPosition) + jacocoSeparator + sourceFilename;
+                // It is a top-leven type that is not the main type. The filename
```
Comment (falko17) [inline Assets/SEE/DataModel/DG/IO/JaCoCoImporter.cs:475]:
```suggestion
                // It is a top-level type that is not the main type. The filename
```

**Review by falko17 — APPROVED** [review]
Just one small comment (probably a new record low from me), so I'll approve already.

File: Assets/SEE/DataModel/DG/IO/JaCoCoImporter.cs
Comment (falko17) [inline Assets/SEE/DataModel/DG/IO/JaCoCoImporter.cs:1]: Oh, one more thing, would it make sense to mark this class as `static`?

Comment (koschke) [reply Assets/SEE/DataModel/DG/IO/JaCoCoImporter.cs:1]: True. Made it static.

**Pull request merged**